components/d2l-applied-filters/d2l-applied-filters.js

```diff
@@ -0,0 +1,218 @@
+				display: inline-block;
+				padding-top: 0.3rem;
+				font-style: italic;
+				color: var(--d2l-color-corundum);
+			}
+		`];
+	}
+
+	static get resources() {
+		return {
+			'ar': {
+			},
+			'de': {
+			},
+			'en': {
+				appliedFilters: 'Applied Filters:',
+				noActiveFilters: 'No active filters'
+			},
+			'es': {
+			},
+			'fi': {
+			},
+			'fr': {
+			},
+			'ja': {
+			},
+			'ko': {
+			},
+			'nb': {
+			},
+			'nl': {
+			},
+			'pt': {
+			},
+			'sv': {
+			},
+			'tr': {
+			},
+			'zh': {
+			},
+			'zh-tw': {
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this._clearSelected = this._clearSelected.bind(this);
+		this._setSelectedOptions = this._setSelectedOptions.bind(this);
+		this._update = this._update.bind(this);
+	}
+
+	firstUpdated() {
+		this._update();
+	}
+
+	updated(changedProperties) {
+		if (Object.keys(changedProperties).indexOf('for') > -1) {
+			this._update();
+		}
+	}
+
+	_update() {
+		this._setOptions();
+		this._setSelectedOptions();
+	}
+
+	_filterOptionChanged() {
+		this._setSelectedOptions();
+	}
+
+	_multiSelectItemDeleted(e) {
+		const entryIndex = Number.parseInt(e.path[0].attributes.index.value);
+		const entry = this._selectedEntries[entryIndex];
+		entry.dispatchEvent(new CustomEvent('d2l-menu-item-select'));
+	}
+
+	_findDropdownIfNested(target) {
+		if (target.nodeName !== DROPDOWN_NAME) {
+			const dropdownChildren = getComposedChildren(target);
+
+			for (let i = 0; i < dropdownChildren.length; i++) {
+				const child = dropdownChildren[i];
+				if (child.nodeName === DROPDOWN_NAME) {
+					return child;
+				}
+				const t = child.querySelector('d2l-filter-dropdown');
+				if (t) {
+					return t;
+				}
+			}
+		}
+		return target;
+	}
+
+	_setFilter() {
+		if (!this.for) {
+			return undefined;
+		}
+		if (this._target) {
+			this._target.removeEventListener('d2l-filter-dropdown-option-change', this._setSelectedOptions, true);
+			this._target.removeEventListener('d2l-filter-dropdown-cleared', this._clearSelected, true);
+			this._target.removeEventListener('d2l-filter-dropdown-category-slotchange', this._update, true);
+		}
+
+		const ownerRoot = this.getRootNode();
+
+		const targetSelector = `#${this.for}`;
+		let target = ownerRoot.querySelector(targetSelector) || (ownerRoot && ownerRoot.host && ownerRoot.host.querySelector(targetSelector));
+
+		if (target) {
+			target = this._findDropdownIfNested(target);
+			target.addEventListener('d2l-filter-dropdown-option-change', this._setSelectedOptions, true);
+			target.addEventListener('d2l-filter-dropdown-cleared', this._clearSelected, true);
+			target.addEventListener('d2l-filter-dropdown-category-slotchange', this._update, true);
+		}
+		this._target = target;
+	}
+
+	_getFilterOptions() {
+		this._setFilter();
+		const dropdown = this._target;
+
+		if (!dropdown) {
+			return [];
+		}
+
+		const results = {};
+		[...dropdown.children]
+			.filter(x => x.nodeName === DROPDOWN_CATEGORY_NAME)
```

Comment:
Is filtering all children going to be an issue.  In portfolio at least we're probably going to have an arbitrary number of tags to apply right?

Could this be a case where state management is warranted?

Reply:
this is only ran when slots are changed on the filter menu, not the multi-select, so in portfolio it runs once with elements there (plus a couple more times when there's no children)

Reply:
Right but if you have a scenario of:

- applied filters and filter menu load (empty)
- Async process to load filters happens
- Each filter is slotted sequentially
- applied filters slotted fires N times iterating over the full children collection of filters

Or is there something smart enough to dedupe those interactions?

Reply:
It's not too hard to only check the changed category, so we could restrict it to only rechecking the category that changed